hex-arch-adapters-spring6/src/main/kotlin/dev/eskt/arch/hex/adapter/spring/EventListenerExecutorService.kt

```diff
@@ -54,34 +57,39 @@ public class EventListenerExecutorService(
-                while (!stopped) {
-                    logger.info("Starting collection of events for $eventListener")
-                    try {
-                        eventStore
-                            .singleStreamTypeEventFlow(
-                                streamType = eventListener.streamType,
-                                sincePosition = bookmark.get(eventListener.id),
-                                batchSize = batchSize,
-                            )
-                            .collect { envelope ->
-                                transactionTemplate.execute {
-                                    eventListener.listen(envelope)
-                                    bookmark.set(eventListener.id, envelope.position)
-                                }
-                                logger.info("Processed event position ${envelope.position} of type ${envelope.event::class.qualifiedName} in $eventListener")
+            run(eventListener)
+        }
+    }
+
+    private fun run(eventListener: SingleStreamTypeEventListener<Any, Any>) {
+        val eventStore = eventStores
+            .singleOrNull { eventListener.streamType in it.registeredTypes }
+            ?: throw IllegalStateException("$eventListener has a stream type which needs to be registered in one (and only one) event store.")
+        val job = scope.launch {
+            while (!stopped) {
+                logger.info("Starting collection of events for $eventListener")
+                try {
+                    eventStore
+                        .singleStreamTypeEventFlow(
+                            streamType = eventListener.streamType,
+                            sincePosition = bookmark.get(eventListener.id),
+                            batchSize = batchSize,
+                        )
+                        .collect { envelope ->
+                            transactionTemplate.execute {
+                                eventListener.listen(envelope)
+                                bookmark.set(eventListener.id, envelope.position)
                             }
-                    } catch (e: Exception) {
-                        if (e is CancellationException) throw e
-                        logger.error("Error while collecting events in $eventListener, will try to restart in $backoff", e)
-                        if (!stopped) delay(backoff)
-                    }
+                            logger.info("Processed event position ${envelope.position} of type ${envelope.event::class.qualifiedName} in $eventListener")
+                        }
+                } catch (e: Exception) {
+                    if (e is CancellationException) throw e
+                    logger.error("Error while collecting events in $eventListener, will try to restart in $backoff", e)
+                    if (!stopped) delay(backoff)
                 }
             }
         }
+        processes[eventListener.id] = EventListenerProcess(job, eventListener)
```

Comment:
I think this function should be called `startJob` and both `init` and `restartEventListener` should keep `processes` up-to-date. They may need to do different things with the map.